### PR TITLE
Add WooPay clickwrap support

### LINKED ISSUE #10122: Add WooPay clickwrap support
If a merchant enables the `I have read and agree to the website terms and conditions *` option via **Appearance > Customize > WooCommerce > Checkout > Terms and conditions page** to force a customer to agree to the site's T&Cs before placing an order...

Markup on 2023-10-18 at 09:36:01

...this button does not appear to render in the WooPay hosted checkout flow:

Markup on 2023-10-18 at 09:37:35

It's possible that this may not be necessary if it's stated somewhere (e.g., _our_ terms of service) that using EPMs like WooPay means that a customer inherently agrees to accept the store's T&Cs, but I couldn't find it. If that isn't the case, then we will want to try to surface this via the `pay.woo.com` page so that a customer has to actively accept the merchant store's T&Cs base on the merchant's store configuration.

### PR DESCRIPTION
**Pull request by leonardola**
#### Changes proposed in this Pull Request
This PR adds support for Clickwrap terms and conditions (checkbox) on WooPay.

See testing instructions on https://github.com/Automattic/woopay/pull/3012

Fixes #10122

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `9752` or branch name `add/2253-clickwrap-terms-and-conditions` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/2253-clickwrap-terms-and-conditions) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 496a5c2822144b4a0e790ecde3952a6f4fd57d01
- Build time: 2024-12-05 16:46:35 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by a-danae — COMMENTED** [review]
Thanks again for your work here, @leonardola ! Jumping back to this after the comment in Automattic/woopay#3012, in case it affects this PR.

File: includes/woopay/class-woopay-session.php [hunk 1]
```diff
@@ -947,12 +949,46 @@ private static function get_option_fields_status() {
...
+	 * @param string $inner_block_name The name of a child block.
+	 * @return array|null
+	 */
+	private static function get_inner_block( $current_block, $inner_block_name ) {
+
+		if ( ! isset( $current_block['innerBlocks'] ) ) {
+			return;
+		}
+
+		$checkout_terms_block_index = array_search(
```
Comment (a-danae) [inline includes/woopay/class-woopay-session.php]: 🤏  nit. Given that this function is a generic `get_inner_block()`, how about renaming this variable to `$inner_block_index`?

It's not a blocker, and things work as expected. On a quick read, I found it confusing whether this method was intended for finding a generic lookup of a block, as the name suggests, or if it was specific for the checkout terms block, as this variable seemed to suggest. I understand this is a generic method, not specific to the checkout terms block.

**Review by a-danae — APPROVED** [review]
👍 Things look good to me.

I left a non-blocker question. Also, left the actual review of testing this PR in Automattic/woopay#3012. Thanks for your work here, @leonardola !

File: includes/woopay/class-woopay-session.php (hunk 1 above)
Comment (leonardola) [reply includes/woopay/class-woopay-session.php]: It's mean to be a generic function. I just forgot to update the variable name after the refactor. Fixed with 496a5c2822144b4a0e790ecde3952a6f4fd57d01

**Comment by leonardola** [comment]
Hey @aheckler this PR changed the label of the `Policies and Custom text` field to `Checkout policies` at /wp-admin > Payments > Settings > WooPay

We've noticed a screenshot of that field [here](https://woocommerce.com/document/woopay-merchant-documentation/#checkout-appearance) could you update the docs when 8.7.0 gets released?

Also now WooPay supports both clickwrap (checkbox) and browserwrap (no checkbox) for the terms and conditions text.

Do you still have the logo you used in screenshots so I can generate new ones for you to update the docs?

**Comment by aheckler** [comment]
> We've noticed a screenshot of that field [here](https://woocommerce.com/document/woopay-merchant-documentation/#checkout-appearance) could you update the docs when 8.7.0 gets released?

Will do!

> Also now WooPay supports both clickwrap (checkbox) and browserwrap (no checkbox) for the terms and conditions text.

Will check on this as well when 8.7.0 is out. :)

> Do you still have the logo you used in screenshots so I can generate new ones for you to update the docs?

I think @csmcneill updated those image slideshows while I was on leave last year, so he may have the logo still. :) If not, we can use any generic logo type image. My test site uses this one:

cat-logo

**Comment by csmcneill** [comment]
> he may have the logo still

I do not! The cat one is better anyway 🐱

**Comment by aheckler** [comment]
This is now documented [here](https://woocommerce.com/document/woopay-merchant-documentation/#checkout-appearance).

**Pull request merged**